contracts/sg-ics721/src/testing/integration_tests.rs

```diff
@@ -0,0 +1,1564 @@
+                test.ics721.clone(),
+                &ExecuteMsg::Callback(CallbackMsg::CreateVouchers {
+                    receiver: "mr-t".to_string(),
+                    create: VoucherCreation {
+                        class: Class {
+                            id: ClassId::new("bad kids"),
+                            uri: Some("https://moonphase.is".to_string()),
+                            // CustomClassData with no owner info
+                            data: Some(
+                                to_binary(&CustomClassData {
+                                    foo: Some(OWNER_SOURCE_CHAIN.to_string()),
+                                })
+                                .unwrap(),
+                            ),
+                        },
+                        tokens: vec![
+                            Token {
+                                id: TokenId::new("1"),
+                                uri: Some("https://moonphase.is/image.svg".to_string()),
+                                data: None,
+                            },
+                            Token {
+                                id: TokenId::new("2"),
+                                uri: Some("https://moonphase.is/image.svg".to_string()),
+                                data: None,
+                            },
+                        ],
+                    },
+                }),
+                &[],
+            )
+            .unwrap();
+        // Check entry added in CLASS_ID_TO_NFT_CONTRACT
+        let nft_contracts = test.query_nft_contracts();
+        assert_eq!(nft_contracts.len(), 1);
+        assert_eq!(nft_contracts[0].0, "bad kids");
+        // Get the address of the instantiated NFT.
+        let nft: Addr = test
+            .app
+            .wrap()
+            .query_wasm_smart(
+                test.ics721.clone(),
+                &QueryMsg::NftContract {
+                    class_id: "bad kids".to_string(),
+                },
+            )
+            .unwrap();
+
+        // check contract info is properly set
+        let contract_info: cw721::ContractInfoResponse = test
+            .app
+            .wrap()
+            .query_wasm_smart(nft.clone(), &Cw721QueryMsg::<Empty>::ContractInfo {})
+            .unwrap();
+        assert_eq!(
+            contract_info,
+            cw721::ContractInfoResponse {
+                name: "bad kids".to_string(),
+                symbol: "bad kids".to_string()
+            }
+        );
+
+        // check collection info is properly set
+        let collection_info: CollectionInfoResponse = test
+            .app
+            .wrap()
+            .query_wasm_smart(nft.clone(), &Sg721QueryMsg::CollectionInfo {})
+            .unwrap();
+
+        assert_eq!(
+            collection_info,
+            CollectionInfoResponse {
+                // creator is ics721 contract, since no owner in ClassData provided
+                creator: test.ics721.to_string(),
```

Comment:
Also here, where custom class data (e.g. from nft-transfer module) has no owner, then ics721 is creator for sg721-base